### PR TITLE
Resolver

### PR DESCRIPTION
**Pull request by vmx**
Implement the "resolver part" of the API.

Please note that an `isLink()` implementation is missing due to https://github.com/ipld/interface-ipld-format/issues/17.

### CODE REVIEW

File: src/resolver.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
+'use strict'
+
+const util = require('./util')
+
+const resolve = (binaryBlob, path, callback) => {
```
Comment (diasdavid) [inline src/resolver.js]: nitpick, you could use `function` declarations here instead of assigning funcs to const.

File: src/util.js [hunk 2]
```diff
@@ -22,14 +22,22 @@ const cid = (dagNode, callback) => {
...
 
+const hashToCid = (hash) => {
+  const multihash = multihashes.encode(hash, 'dbl-sha2-256')
+  const cidVersion = 1
+  const cid = new CID(cidVersion, 'bitcoin-block', multihash)
+  return cid
+}
+
 module.exports = {
+  hashToCid: hashToCid,
```
Comment (diasdavid) [inline src/util.js]: Needs to be documented

File: src/resolver.js (hunk 1 above)
Comment (vmx) [reply src/resolver.js]: I personally prefer `var functionname = function(bla) {}` style over `function functioname(bla)`. But if the general style is `function functioname(bla)` I'm happy to change that in the whole code base.

Comment (diasdavid) [reply src/resolver.js]: Right now we don't have a fixed style although I do write and push more for the `function name ()`. Let's keep it as it is then.

File: src/util.js (hunk 2 above)
Comment (diasdavid) [reply src/util.js]: Any plan to get this documented in the README as well?

Comment (vmx) [reply src/util.js]: That will only get API docs. It's not really public API, it just leaked into it because of the way JS works.

**Pull request merged**